router/batchrouter/batchrouter.go

```diff
@@ -1526,9 +1550,24 @@ func (worker *workerT) workerProcess() {
 					brtQueryStat := stats.NewTaggedStat("batch_router.jobsdb_query_time", stats.TimerType, map[string]string{"function": "workerProcess"})
 					brtQueryStat.Start()
 					brt.logger.Debugf("BRT: %s: DB about to read for parameter Filters: %v ", brt.destType, parameterFilters)
-
-					retryList := brt.jobsDB.GetToRetry(jobsdb.GetQueryParamsT{CustomValFilters: []string{brt.destType}, JobCount: toQuery, ParameterFilters: parameterFilters, IgnoreCustomValFiltersInQuery: true})
-					unprocessedList := brt.jobsDB.GetUnprocessed(jobsdb.GetQueryParamsT{CustomValFilters: []string{brt.destType}, JobCount: toQuery, ParameterFilters: parameterFilters, IgnoreCustomValFiltersInQuery: true})
+					retryList := brt.jobsDB.GetToRetry(
+						jobsdb.GetQueryParamsT{
+							CustomValFilters:              []string{brt.destType},
+							JobsLimit:                     toQuery,
+							ParameterFilters:              parameterFilters,
+							IgnoreCustomValFiltersInQuery: true,
```

Comment:
Not sure about 1 and 2, but params should take care of part 3 if `readPerDestination` is set

Reply:
If this is a query optimisation, e.g. don't include the `customVal` filter if a `destinationId` parameter filter is provided, then we could implement this in a better way, automatically so that it applies in all queries and we can drop this flag from the params

Reply:
1 is done for jobsdb cache to work properly I believe.

Reply:
Inside the cache we appear to be using `CustomValFilter` as a key and ignore the existence of `IgnoreCustomValFiltersInQuery`. 

Isn't the following scenario problematic?

1. Query using {CustomValFilters: "val"} returns 0 jobs and is cached.
2. Next query using {CustomValFilters: "val", IgnoreCustomValFiltersInQuery: true} would return some jobs if it went through the database but due to the cache we reply with 0 jobs.